### PR TITLE
Handling Integrals containing I

### LINKED ISSUE #17841: integrate throws error for rational functions involving I
Relatively simple integrals such as
```python
x = symbols('x')
f = diff(1/(x**2+x+I),x)
integrate(f,x)
```
fail with the error
```
sympy.polys.polyerrors.PolynomialDivisionFailed: couldn't reduce degree in a polynomial division algorithm when dividing [EX(-2), EX(-1)] by [EX(-(-15 + 8*I)*(15 + 8*I)/289)]. This can happen when it's not possible to detect zero in the coefficient domain. The domain of computation is EX. You may want to use a different simplification algorithm. Note that in general it's not possible to guarantee to detect zero in this domain.
```
The best workaround I have found so far is
```python
c = symbols('c',complex=True)
integrate(f.subs(I,c),x).subs(c,I)
```
but ``integrate(f,x,manual=True)`` also works.

### PR DESCRIPTION
**Pull request by Smit-create**
Handling Integrals containing I
#### References to other Issues or PRs

Fixes #17841

#### Brief description of what is fixed or changed
Added additional `expand` step in `simplify` method of `ExpressionDomain`

#### Release Notes

* polys
  * Added  `expand` in `simplify` method of `ExpressionDomain`

### CODE REVIEW

File: sympy/polys/domains/expressiondomain.py [hunk 1]
```diff
@@ -9,6 +9,9 @@
 from sympy.polys.polyutils import PicklableWithSlots
 from sympy.utilities import public
 
+eflags = dict(mul=True, power_exp=False, power_base=False,
```
Comment (jksuom) [inline sympy/polys/domains/expressiondomain.py]: The first attribute could be `deep=False` to save time. It is not necessary to inspect anything but the topmost node as this `simplify` method will be called repeatedly on each step.

**Comment by jksuom** [comment]
It seems that integral tests are sensitive to format changes in expressions.  Polynomials with domain EX should be avoided if possible. This patch appears to help:
```
--- a/sympy/integrals/rationaltools.py
+++ b/sympy/integrals/rationaltools.py
@@ -212,7 +212,7 @@ def ratint_logpart(f, g, x, t=None):
     a, b = g, f - g.diff()*Poly(t, x)
 
     res, R = resultant(a, b, includePRS=True)
-    res = Poly(res, t, composite=False)
+    res = Poly(res, t)
 
     assert res, "BUG: resultant(%s, %s) can't be zero" % (a, b)
 ```

**Comment by jksuom** [comment]
It looks like the removal of `composite=False` makes the slow Wester test fail and has to be reverted. Actually, the code should work with different formats but some condition tests fail in the expression domain. You could try this fix:
```
--- a/sympy/polys/domains/expressiondomain.py
+++ b/sympy/polys/domains/expressiondomain.py
@@ -203,7 +203,7 @@ def is_positive(self, a):
 
     def is_negative(self, a):
         """Returns True if ``a`` is negative. """
-        return a.ex.as_coeff_mul()[0].is_negative
+        return a.ex.could_extract_minus_sign()
 
     def is_nonpositive(self, a):
         """Returns True if ``a`` is non-positive. """
```

**Comment by jksuom** [comment]
This seems to work. For the record, the current implementation is failing because the results of `EX.is_negative()` differ on non-expanded and expanded arguments.
```
>>> from sympy import *
>>> x = symbols('x')
>>> k, a0 = symbols('k a0', real=True)
>>> e = -(a0**4 + 2*a0**2*k**2 + k**4)/(2*a0**2 + 2)
>>> EX.is_negative(EX(e))
False
>>> EX.is_negative(EX(e.expand()))
False
>>> EX.is_negative(EX(-e))
True
>>> EX.is_negative(EX(-e.expand()))
False
```
This should be properly investigated.

**Comment by Smit-create** [comment]
> . For the record, the current implementation is failing because the results of `EX.is_negative()` differ on non-expanded and expanded arguments.

On the current branch i get:
```
>>> from sympy import *
>>> x=symbols('x')
>>> k, a0 = symbols('k a0', real=True)
>>> e = -(a0**4 + 2*a0**2*k**2 + k**4)/(2*a0**2 + 2)
>>> EX.is_negative(EX(e))
True
>>> EX.is_negative(EX(e.expand()))
True
>>> EX.is_negative(EX(-e))
False
>>> EX.is_negative(EX(-e.expand()))
False
```

**Comment by Smit-create** [comment]
@jksuom Is this ready to merge?

**Comment by jksuom** [comment]
Thanks, I think so. The comparison methods of `EX` domain seem very fragile, but I cannot think of a better fix now. I still believe that `composite=False` in `rationaltools` should be removed to avoid the `EX` domain but the reason for failure of `test_W23` should be found first.

**Comment by Smit-create** [comment]
Yes, I will try to find it out. Thanks :)

**Pull request merged**